src/lib/utils/settings_manager.cpp

```diff
@@ -0,0 +1,36 @@
+#include "settings_manager.hpp"
+
+#include "utils/settings/mock_setting.hpp"
+
+namespace opossum {
+
+bool SettingsManager::has_setting(const std::string& name) const { return _settings.count(name); }
+
+void SettingsManager::add(const std::string& name, std::shared_ptr<AbstractSetting> setting) {
```

Comment:
Something around here seems not to work, maybe because of using the attribute of the setting as the key?

Reply:
What exactly is not working?

Reply:
I sometimes get std::bad_alloc or segfaults when accessing or adding settings.

Reply:
e.g. hyriseTest built with clang
```
[ RUN      ] MetaSettingsTest.IsUpdateable
unknown file: Failure
C++ exception with description "std::bad_alloc" thrown in SetUp().
```
where this is works with gcc (but there the names are not set correctly).

Reply:
Sounds like memory corruption to me. I'd try asan/ubsan.

Reply:
Yep, was absolutely stupid. Stored `std::string& name` as AbstractSetting member.